### ISSUE TITLE
[Aspire] First time running `dotnet msbuild --getProperty:IsAspireHost` fails due to telemetry notice

### ISSUE THREAD

**Issue by vhvb1989** [Azure/azure-dev#4513]
Usually, when running `dotnet msbuild --getProperty:IsAspireHost` , the response is either `false` or `true`. Azd uses this output to detect an Aspire project.

However, on a very fresh dotnet installation, the first response from `dotnet msbuild --getProperty:IsAspireHost` includes a telemetry notice like:

```
PS C:\Users\vivazqu\t2\eShop_dyigitbas\src\eShop.AppHost> dotnet msbuild --getProperty:IsAspireHost

This is .NET 9.0.
---------------------
SDK Version: 9.0.100-rc.2.24474.11

Telemetry
---------
The .NET tools collect usage data to help us improve your experience. Microsoft collects this data and shares it with the community. You can opt out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to "1" or "true" using your favorite shell.

Read more about .NET CLI tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Install an ASP.NET Core HTTPS development certificate.
To trust the certificate, run "dotnet dev-certs https --trust"
Learn about HTTPS: https://aka.ms/dotnet-https

----------------
Write your first application: https://aka.ms/dotnet-hello-world
Discover what's new: https://aka.ms/dotnet-whats-new
Explore the documentation: https://aka.ms/dotnet-docs
Report issues and find the source code on GitHub: https://github.com/dotnet/core
Use "dotnet --help" to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
true
```

This output makes `azd` to think the project is not Aspire.

I was able to reproduce by installing dotnet 9 on a fresh new Windows instance and running `dotnet msbuild --getProperty:IsAspireHost` inside an AppHost project.  Here is some pictures (the VM was in Spanish, but it should help to illustrate the issue)

![Image](https://github.com/user-attachments/assets/9e4764bd-4797-4f1f-a216-c54769c9a0e5)

// Note how, from the second time ahead, the output is the expected.

**Comment by vhvb1989** [comment]
maybe: https://github.com/dotnet/sdk/issues/3828

**Comment by vhvb1989** [comment]
https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#dotnet_skip_first_time_experience

**Comment by rajeshkamal5050** [comment]
@vhvb1989 is this must-fix for 1.10.4? Adding it to backlog. Please pull it into the release if need be.
